### PR TITLE
Add spec tests for api_fetch function

### PR DESCRIPTION
**Pull request by marek130**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -35,6 +35,7 @@ group :development do
   gem "rubocop-rspec", '= 2.0.1',                      require: false
   gem "rb-readline", '= 0.5.5',                        require: false, platforms: [:mswin, :mingw, :x64_mingw]
   gem "github_changelog_generator", '~> 1.15',         require: false if Gem::Version.new(RUBY_VERSION.dup) >= Gem::Version.new('2.3.0')
+  gem "webmock",                                       require: false
```
Comment (anders-larsson) [inline Gemfile:38]: gem should probably be added to .sync.yml too to make sure it survives 'pdk update' command.

File: spec/classes/data_types_spec.rb [hunk 2]
```diff
@@ -174,7 +182,15 @@
             context "when #{var_name} (#{type}) is set to valid #{valid} (as #{valid.class})" do
               let(:params) { [mandatory_params, var[:params], { "#{var_name}": valid, }].reduce(:merge) }
 
-              it { is_expected.to compile }
+              it do
```
Comment (anders-larsson) [inline spec/classes/data_types_spec.rb:185]: We probably want to have tests as well on what happens if we do not get a proper answer from function.

Comment (anders-larsson) [reply spec/classes/data_types_spec.rb:185]: Maybe this is applicable in a different spec file though. Maybe these tests already exists? Don't remember :P

**Review by anders-larsson — COMMENTED** [review]
Looks like a good start. We probably want something related to timeouts and other issues as well to make sure we understand how that is handled.

File: Gemfile (hunk 1 above)
Comment (marek130) [reply Gemfile:38]: .sync.yml file updated

File: spec/classes/data_types_spec.rb (hunk 2 above)
Comment (marek130) [reply spec/classes/data_types_spec.rb:185]: yes, these use cases are included in spec/classes/init_spec.rb

**Comment by marek130** [comment]
> Looks like a good start. We probably want something related to timeouts and other issues as well to make sure we understand how that is handled.

Added new test use case which checks timeout

**Comment by anders-larsson** [comment]
I'll merge these changes. Thanks.

**Pull request merged**